kuma/wiki/views/code.py

```diff
@@ -18,6 +20,10 @@
 @allow_CORS_GET
 @xframe_options_exempt
 @process_document_path
+@cache_control(
+    public=True,
+    max_age=settings.CODE_SAMPLES_CACHE_CONTROL_MAX_AGE
+)
```

Comment:
This would work. However, I agree an ETag would be better, and would cache changes in the template as well.

I don't think I'm ready to set ``USE_TAGS`` yet, but I could be convinced. [From the docs](https://docs.djangoproject.com/en/1.11/ref/middleware/#django.middleware.common.CommonMiddleware), it looks like the processing of ETags moved from ``CommonMiddleware`` to the ``ConditionalGetMiddleware``, so we may want to wait until 1.11 to enable this across the site. Or maybe we're ready.

If we just want to use an ETag on this view, I think it would be easy to create a decorator based on Django 1.8's ``CommonMiddleware``:

https://github.com/django/django/blob/2d4bc5a60aa8a076689667c550ded96b87bc463e/django/middleware/common.py#L113-L127

I agree we do the same amount of work, however, so the ETag isn't strictly needed.  An alternate would be to hash the revision ID alone as the ETag, and add the "complete" ETag if and when we update the base template.

Or, just keep it the way you did it the first time.